### PR TITLE
IDEM-210

### PR DESCRIPTION
**Pull request by xiangzhe0812**
sync detail from proxy

### CODE REVIEW

**Review by bitcoinbrisbane — APPROVED** [review]

File: src/services/ThirdPartyService.ts [hunk 1]
```diff
@@ -127,4 +133,42 @@ export class ThirdPartyService implements IThirdPartyService {
...
+                    {
+                        headers: {
+                            "Content-Type": "application/json",
+                            Authorization: `Bearer ${token}`
+                        }
+                    }
+                );
+            }
+        }
+    }
```
Comment (pondpiu) [inline src/services/ThirdPartyService.ts:173]: should we throw or return error in unexpected case (else case of the two ifs) ?

File: src/user/user.controller.ts [hunk 2]
```diff
@@ -64,6 +65,7 @@ export class UserController {
         @Body() body: UserVerifyRequestBody
     ): Promise<KycResponse> {
         let user: User;
+        console.log(body);
```
Comment (pondpiu) [inline src/user/user.controller.ts:68]: log

**Review by bitcoinbrisbane — DISMISSED** [review]

File: src/services/ThirdPartyService.ts [hunk 3]
```diff
@@ -127,4 +133,53 @@ export class ThirdPartyService implements IThirdPartyService {
...
+                        headers: {
+                            "Content-Type": "application/json"
+                        }
+                    }
+                )
+                .catch((error: AxiosError) => {
+                    this.logger.error(error);
+                    throw new Error(error.message);
+                });
+            if (authentication.status === 200) {
```
Comment (pondpiu) [inline src/services/ThirdPartyService.ts:159]: throw error in else

File: src/services/ThirdPartyService.ts [hunk 4]
```diff
@@ -127,4 +133,53 @@ export class ThirdPartyService implements IThirdPartyService {
             throw new Error(error.response.data);
         }
     }
+
+    public async syncDetail(userDetail: UserDetailRequest): Promise<void> {
+        let endPoint: string;
+        if (userDetail.source === VendorEnum.GPIB) {
```
Comment (pondpiu) [inline src/services/ThirdPartyService.ts:139]: throw error in else

**Review by pondpiu — APPROVED** [review]

**Review by JasminFarhi — APPROVED** [review]

**Pull request merged**